### PR TITLE
Add mono-error docs.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

**Comment by kumpera** [comment]
@akoeplinger does this look right?

File: docs/advanced/runtime/docs/index.md [hunk 1]
```diff
@@ -24,6 +24,14 @@ This page currently contains links to the wiki-fied version of the runtime docum
...
 -   [Glossary](/docs/advanced/runtime/docs/glossary)
 
+### Runtime guidelines
+
+The following describe guidelines for specific areas of the runtime
+
+-   [Error handling/MonoError](/docs/advanced/runtime/docs/mono-error/)
+
+
+
```
Comment (akoeplinger) [inline docs/advanced/runtime/docs/index.md:34]: Nit: use only one newline here. I sometimes run the repo through a markdown linter and it complains about that :)

File: docs/advanced/runtime/docs/mono-error.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+Current implementation
+======================
+
+The current implementation exists in mono-error.h and mono-error-internals.h. The split is so API users can consume errors, but they
+are not supported to be able to produce them - such use case has yet to arise.
+
+### Writing a function that produces errors
+
+```
```
Comment (akoeplinger) [inline docs/advanced/runtime/docs/mono-error.md]: You can add `c` after the backticks and we'll get a nice syntax highlighting on the website.

File: docs/advanced/runtime/docs/mono-error.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+- Add a "MonoError *error" argument as the last to your function
+- Call "mono_error_init (error)" unconditionally at the beginning
+- Call one of the mono_error_set functions based on what managed exception this should produce and the available information
+- Document that a NULL returns means an error
+
+
+Writing a function that consumes errors
+=======================================
+
+```
```
Comment (akoeplinger) [inline docs/advanced/runtime/docs/mono-error.md]: same here

**Comment by akoeplinger** [comment]
@kumpera Looks fine

**Comment by kumpera** [comment]
Fixed the issues pointed above. merging.

**Pull request merged**